zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -728,8 +1031,14 @@ private String create(final String path, final Object dataObject, final List<ACL
       final byte[] dataBytes = dataObject == null ? null : serialize(dataObject, path);
       checkDataSizeLimit(path, dataBytes);
 
-      final String actualPath = retryUntilConnected(
-          () -> getExpectedZookeeper(expectedSessionId).create(path, dataBytes, acl, mode));
+      final String actualPath;
+      if (mode.isTTL()) {
+        actualPath = retryUntilConnected(() -> getExpectedZookeeper(expectedSessionId)
+            .create(path, dataBytes, acl, mode, new Stat(), ttl));
```

Comment:
Do we want a new `Stat` or `null` for this? We're not doing anything with the stat object, right?

Reply:
Yes, that makes sense. I changed it to `null`.